arch/common/rom_start_offset.ld

```diff
@@ -4,5 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+KEEP(*(.rom_start_area))
+KEEP(*(".rom_start_area.*"))
+
```

Comment:
This is not needed.

First, `CONFIG_ROM_OFFSET` used below is defined as:
https://github.com/zephyrproject-rtos/zephyr/blob/45e1ff94cdbc395ab9f87d948580cefd585479c5/Kconfig.zephyr#L127-L140

which then is no longer true on a regular Zephyr build.

I understand that certain users / applications may have special needs and for various needs must inject code before.

And for this we have `zephyr_linker_sources()`, which is already used to include this file, as seen:
https://github.com/zephyrproject-rtos/zephyr/blob/45e1ff94cdbc395ab9f87d948580cefd585479c5/arch/common/CMakeLists.txt#L55

So to get your own code inject **before** this line, you can use:
```
  zephyr_linker_sources(ROM_START SORT_KEY -0x0 i_need_this_first.ld)
```
in which case users are having full control, and thus must of course be aware that they have customized the build.